internal/distro/fedora/distro.go

```diff
@@ -704,12 +705,29 @@ func (t *imageType) initializeManifest(bp *blueprint.Blueprint,
 	/* #nosec G404 */
 	rng := rand.New(source)
 
+	// the os pipeline filters repos based on the `osPkgsKey` package set, merge the repos which
+	// contain a payload package set into the `osPkgsKey`, so those repos are included when
+	// building the rpm stage in the os pipeline
```

Comment:
This is unfortunately not the right approach.

We do not add payload repositories intentionally under the `osPkgsKey` key to ensure that only the "system" repos are used when depsolving the base image pkg set. And only on top of that already depsolved set, we add user-requested packages and also add the payload repos so the list. This assures that the user repo content can't replace any of the base packages, but can be used for any content on top of that.

If the user wants to use their own repos for the base package set, then they need to override system repositories in `/etc/osbuild-composer/repositories`. This is however possible only on-premise.

This change would allow the user-provided payload repos to be used for image base pkg set even in the service. So someone could potentially provide some other EL-clone repos and have the RHEL image be built only using that content (if versions would be higher than what is in RHEL repos, since we do not set hugher priorities for user repos at the moment).

Reply:
This happens after depsolving though. Only when instantiating the manifest the 2nd time do they get added. But I totally agree that if we can solve it in the manifest package that would be much, much cleaner.

Reply:
Thinking about this a bit more, we should probably make sure that the Workload repos are taken into account for any pipeline implementation from the `internal/manifest` package.

From a brief look:

```bash
thozza@fedora ~/devel/osbuild-composer/internal/manifest (main =)
$ grep -r 'NewRPMStageOptions('
anaconda.go:    pipeline.AddStage(osbuild.NewRPMStage(osbuild.NewRPMStageOptions(p.repos), osbuild.NewRpmStageSourceFilesInputs(p.packageSpecs)))
build.go:       pipeline.AddStage(osbuild.NewRPMStage(osbuild.NewRPMStageOptions(p.repos), osbuild.NewRpmStageSourceFilesInputs(p.packageSpecs)))
commit_server_tree.go:  pipeline.AddStage(osbuild.NewRPMStage(osbuild.NewRPMStageOptions(p.repos), osbuild.NewRpmStageSourceFilesInputs(p.packageSpecs)))
coreos_installer.go:    pipeline.AddStage(osbuild.NewRPMStage(osbuild.NewRPMStageOptions(p.repos), osbuild.NewRpmStageSourceFilesInputs(p.packageSpecs)))
os.go:  rpmOptions := osbuild.NewRPMStageOptions(p.repos)
```

Reply:
As discussed offline, the workload is only properly instantiated during the first `initializeManifest` call from `PackageSets`, as they're based on these pacakge sets. The 2nd invocation doesn't have the needed information to construct the workload the same way. 

I'm not sure what the best approach is, I moved the repo juggling to `Manifest`, instead of `initializeManifest`, which I think is correct, but ugly. Another option is to construct the workloads differently somehow if PackageSets is nil.